src/run_speedy.jl

```diff
@@ -1,50 +1,47 @@
 
-    # initialize the model setup, precalculate constants
-    P = Parameters(NF=NF,kwargs...)
-    C = Constants(P)
-    G = GeoSpectral(P)
-    B = Boundaries(P)
-    HD = HorizontalDiffusion(P,C,G,B)
-    M = Model(P,C,G,B,HD)               # pack all of the above into a Model struct
+    # INITALIZE MODEL
+    prog_vars,diag_vars,model_setup = initialize_model(NF,kwargs...)
 
-    # initialize variables
-    Prog = initial_conditions(P,B,G)
-    # Diag = DiagnosticVariables(P)
-
-    # start speedy model integration
-    # time_stepping!(Prog,Diag,M)
-
-    return Prog
+    # START MODEL INTEGRATION
+    time_stepping!(prog_vars,diag_vars,model_setup) 
+    return prog_vars                                # return prognostic variables when finished
 end
 
-"temporary function, testing out different structure"
-function initialize_model(::Type{NF}=Float64;         # number format, use Float64 as default
-                          kwargs...                   # all additional non-default parameters
+"""
+    prog_vars, diag_vars, model_setup = initialize_model(NF,kwargs...)
+    
+Initialize the model by returning
+- `prog_vars`, the initial conditions of the prognostic variables
+- `diag_vars`, the preallocated the diagnotic variables (initialised to zero)
+- `model_setup`, the collected pre-calculated structs that don't change throughout integration:
+parametes, constants, geometry, spectral transform, boundaries, diffusion."""
+function initialize_model(::Type{NF}=Float64;       # number format, use Float64 as default
+                          kwargs...                 # all additional non-default parameters
                           ) where {NF<:AbstractFloat}
 
-    P   = Parameters(NF=NF,kwargs...)
-    C   = Constants(P)
-    G   = GeoSpectral(P)
-    B   = Boundaries(P)
-    HD  = HorizontalDiffusion(P,C,G,B)
-    M   = Model(P,C,G,B,HD)
-
-    DiagnosticVars = DiagnosticVariables{NF}(G)
-    PrognosticVars = initial_conditions(P,B,G)
-
-    return PrognosticVars,DiagnosticVars,M
+    P = Parameters(NF=NF,kwargs...)                 # all model parameters chosen through kwargs
+    C = Constants(P)                                # constants used in model integration
+    G = GeoSpectral(P)                              # geometry and spectral transform structs
+    B = Boundaries(P)                               # arrays for boundary conditions
+    H = HorizontalDiffusion(P,C,G,B)                # precomputed arrays for horizontal diffusion
+    M = ModelSetup(P,C,G,B,H)                       # pack all of the above into a ModelSetup struct
+
+    prognostic_vars = initial_conditions(P,B,G)     # initialize prognostic variables
+    diagnostic_vars = DiagnosticVariables(G)        # preallocate all diagnostic variables with zeros
+    
+    return prognostic_vars,diagnostic_vars,M
 end
 
 """Struct holding all the model structs """
-struct Model{NF<:AbstractFloat}
+struct ModelSetup{NF<:AbstractFloat}
```

Comment:
I think I'd like to call it `ModelSetup` instead to make clear it's only the setup of the model, not the model state (which is defined with the prognostic variabels) nor the diagnostic variables. `ModelSetup` is essentially every "constant" whether it's a scalar or an array that doesn't change through model integration.